test/unit/Elsa.Activities.UnitTests/HTTP/SendHttpRequestTests.cs

```diff
@@ -0,0 +1,390 @@
+            Method = new Input<string>("GET"),
+            ExpectedStatusCodes = new List<HttpStatusCodeCase>(),
+            Timeout = mockActivity
+        };
+
+        // Assert - This tests property configuration
+        Assert.Equal(mockActivity, sendHttpRequest.Timeout);
+    }
+
+    [Fact]
+    public void Should_Allow_Configuration_Of_UnmatchedStatusCode_Activity()
+    {
+        // Arrange
+        var mockActivity = Substitute.For<IActivity>();
+        var url = "https://api.example.com/error";
+        
+        // Act
+        var sendHttpRequest = new SendHttpRequest
+        {
+            Url = new Input<Uri?>(new Uri(url)),
+            Method = new Input<string>("GET"),
+            ExpectedStatusCodes = new List<HttpStatusCodeCase>(),
+            UnmatchedStatusCode = mockActivity
+        };
+
+        // Assert - This tests property configuration
+        Assert.Equal(mockActivity, sendHttpRequest.UnmatchedStatusCode);
+    }
+
+    [Fact]
+    public async Task Should_Schedule_No_Activity_When_No_Status_Code_Cases_Match_And_No_Unmatched_Handler()
+    {
+        // Arrange
+        var (configured, children) = CreateSendHttpRequestWithStatusHandlers([(200, "handler200")], unmatchedHandler: null); 
+        var responseHandler = CreateResponseHandler(HttpStatusCode.InternalServerError); // 500 - no match
+
+        // Act
+        var context = await ExecuteActivityWithScheduling(configured, responseHandler, children.Values.ToArray());
+
+        // Assert - Verify that no activities were scheduled since there's no handler for this status code
+        // Although child activities were passed to the workflow, none are scheduled because there is no matching status code case and no unmatched handler.
+        var allScheduledActivities = context.WorkflowExecutionContext.Scheduler.List().ToList();
+        Assert.Empty(allScheduledActivities);
+    }
+
+    [Fact]
+    public async Task Should_Set_Response_Headers_Output()
+    {
+        // Arrange
+        var expectedHeaders = new Dictionary<string, string>
+        {
+            { "Custom-Header", "CustomValue" },
+            { "X-Rate-Limit", "100" }
+        };
+
+        var responseHandler = CreateResponseHandler(HttpStatusCode.OK, additionalHeaders: expectedHeaders);
+        var sendHttpRequest = CreateSendHttpRequest(new Uri("https://api.example.com/headers"));
+
+        // Act
+        var context = await ExecuteActivityWithResponseHandler(sendHttpRequest, responseHandler);
+
+        // Assert
+        var responseHeadersObj = context.GetExecutionOutput(_ => sendHttpRequest.ResponseHeaders);
+        var responseHeaders = responseHeadersObj as HttpHeaders;
+        Assert.NotNull(responseHeaders);
+        Assert.True(responseHeaders.ContainsKey("Custom-Header"));
+        Assert.True(responseHeaders.ContainsKey("X-Rate-Limit"));
+    }
+
+    [Fact]
+    public void Should_Have_Correct_Activity_Attributes_And_HttpStatusCodeCase_Creation()
```

Comment:
A test should focus on one thing, but this test is testing two things:

1. Validating metadata
2. Testing assignment of an activity

I would remove the second test, since that only tests property assignment of an auto-generated property that doesn't contain any logic. After that, I'd rename the method to just `Should_Have_Correct_Activity_Attributes`